plugins/chat/app/policies/chat/direct_message_channel/can_message_all_target_users_policy.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+class Chat::DirectMessageChannel::CanMessageAllTargetUsersPolicy < Chat::DirectMessageChannel::NotPreventingMessagesFromAnyTargetUsersPolicy
```

Comment:
As these policies all do the same thing, we can inherit and just redefine the specific conditions. Also here I’ve kept the approach of having several small policies, but with these policy objects we could have one big policy and just set the proper reason when it fails.

Reply:
Very nice -- in this case I think it's slightly more confusing to inherit from `NotPreventingMessagesFromAnyTargetUsersPolicy` and doesn't skip too much work (since it's just the delegate that needs to be added for each class), and also I find it a bit strange to use `NotPreventingMessagesFromAnyTargetUsersPolicy` as a standalone policy check in the service too.

My preference here would just be to make a unique policy for all the `NotPreventingMessagesFromAnyTargetUsersPolicy` inheriting classes with just a bit more duplication.